enclave/core/sgx/dlmalloc/dlmalloc_wrapper.c

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <openenclave/bits/safecrt.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/enclavelibc.h>
+#include <openenclave/internal/fault.h>
+#include <openenclave/internal/globals.h>
+#include <openenclave/internal/malloc.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/thread.h>
+
+#define HAVE_MMAP 0
+#define LACKS_UNISTD_H
+#define LACKS_SYS_PARAM_H
+#define LACKS_SYS_TYPES_H
+#define LACKS_TIME_H
+#define MORECORE sbrk
+#define ABORT oe_abort()
+#define USE_DL_PREFIX
+#define LACKS_STDLIB_H
+#define LACKS_STRING_H
+#define USE_LOCKS 1
+#define size_t size_t
+#define ptrdiff_t ptrdiff_t
+#define memset oe_memset
+#define memcpy oe_memcpy
+#define sbrk oe_sbrk
+#define fprintf _dlmalloc_stats_fprintf
+
+typedef struct _FILE FILE;
+
+static int _dlmalloc_stats_fprintf(FILE* stream, const char* format, ...);
+
+#pragma GCC diagnostic push
+#ifdef __clang__
+#pragma GCC diagnostic ignored "-Wparentheses-equality"
+#endif
+
+/* Indirection to alloc other allocators */
+#define dlmalloc         _oe_malloc
+#define dlcalloc         _oe_calloc
+#define dlrealloc        _oe_realloc
+#define dlmemalign       _oe_memalign
+#define dlposix_memalign _oe_posix_memalign
+#define dlfree           _oe_free
```

Comment:
All these should have the leading underscore dropped as mentioned above.

Reply:
As for the previous comment, these are separate symbols.